### DISCUSSION TITLE
Get undefined

### DISCUSSION THREAD

**Discussion by ashtonmths** [eritislami/evobot#575] [Q&A]
I get this when I execute an command
```Cannot Read Property 'Get' of Undefined```

**Comment by evrardco** [comment]
Can you you tell us which command? What exactly did you do to get this error? Also, please consider making an issue about (and read the guidelines to doing so).

**Reply by ashtonmths** [reply]
All the commands. I mean the commands which has ```.get(message.guild.id);```. Do we need to setup mongoose?

**Reply by evrardco** [reply]
Hmm. It sounds like that perhaps the bot doesn't have the rights to read the message of the server ? Can you try giving it the admin rights ? To see if the problem lies within the bot's rights or elsewhere.

**Reply by ashtonmths** [reply]
here u go. the error. i ve removed the config file and have stored the values in the util file directly. and i use this command handler called easy djs commandhandler so yeah.
````
UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'get' of undefined
    at Command.run (D:\project1\commands\Music\play.js:21:50)
    at CommandHandler.handle (D:\project1\node_modules\easy-djs-commandhandler\src\index.js:160:8)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**Reply by ashtonmths** [reply]
> Hmm. It sounds like that perhaps the bot doesn't have the rights to read the message of the server ? Can you try giving it the admin rights ? To see if the problem lies within the bot's rights or elsewhere.

yep it has the ADMINISTRATOR privilege. but the issue still persists

**Reply by ashtonmths** [reply]
here is the code thats getting the error
`const serverQueue = message.client.queue.get(message.guild.id);`

**Reply by ashtonmths** [reply]
@evrardco anything?
I removed the await functions cuz it weren't in the async function.

**Reply by maali69** [reply]
I would like to know the code that auto delete the bot's messages. e.g "the music player auto deletes its self after the song ends".

**Reply by ashtonmths** [reply]
> I would like to know the code that auto delete the bot's messages. e.g "the music player auto deletes its self after the song ends".

Pls open a new discussion

**Reply by evrardco** [reply]
@Subtilizer28  I'll give it a look when I have a bit more time. I'm intrigued.
@maali69  Good for you, now stop spamming please.

**Reply by ashtonmths** [reply]
> @Subtilizer28 I'll give it a look when I have a bit more time. I'm intrigued.
> @maali69 Good for you, now stop spamming please.

OH thanks for that!! Ping me once u find an fix. I'll try to fix it too! Thanks!

**Reply by maali69** [reply]
@Subtilizer28 
i have started but no one answered it? name of discussion: "code"
can u answer it please @Subtilizer28 @evrardco

**Reply by maali69** [reply]
@Subtilizer28 @evrardco 
??

**Reply by ashtonmths** [reply]
> @Subtilizer28 @evrardcl(don't wanna ping him cuz it would be a disturbance)
> ??

Just be patient. If u are gonna go on pinging everyone in all discussions then no one would help u. So just be patient

**Reply by maali69** [reply]
are you looking for it i will be so thankfull if u can find it please. i am pinging you last time please @Subtilizer28

**Reply by maali69** [reply]
@evrardco can u please find it

**Reply by ashtonmths** [reply]
@evrardco yo I fixed this issue. I'll open another discussion cuz Ive got another issue. Thanks!!

**Reply by ashtonmths** [reply]
Where to close this discussion tho

**Reply by evrardco** [reply]
@Subtilizer28 What was it? :)

**Reply by ashtonmths** [reply]
@evrardco at this part I get cannot read property url of undefined in play.js command file
```
try {
        const results = await youtube.searchVideos(search, 1, { part: "snippet" });
        songInfo = await ytdl.getInfo(results[0].url);
        song = {
          title: songInfo.videoDetails.title,
          url: songInfo.videoDetails.video_url,
          duration: songInfo.videoDetails.lengthSeconds
        }; 
```

**Reply by ashtonmths** [reply]
In the 3rd line

**Reply by ashtonmths** [reply]
@evrardco umm anything?

**Reply by Takiyo0** [reply]
that try & catch should catch the error. `cannot read property url of undefined` mean that there's no data inside `results` so that mean there's no song was found from `const results = await youtube.searchVideos(search, 1, { part: "snippet" });`
